src/node/in_memory.rs

```diff
@@ -699,23 +698,20 @@ impl<S: std::fmt::Debug + ForkSource> InMemoryNodeInner<S> {
         mut l2_tx: L2Tx,
         gas_per_pubdata_byte: u64,
         tx_gas_limit: u32,
-        l1_gas_price: u64,
-        mut batch_env: L1BatchEnv,
+        batch_env: L1BatchEnv,
         system_env: SystemEnv,
         fork_storage: &ForkStorage<S>,
     ) -> VmExecutionResultAndLogs {
         let tx: Transaction = l2_tx.clone().into();
-        let l1_gas_price =
-            adjust_l1_gas_price_for_tx(l1_gas_price, L2_GAS_PRICE, tx.gas_per_pubdata_byte_limit());
 
-        let coefficients = OverheadCoefficients::from_tx_type(EIP_712_TX_TYPE);
         // Set gas_limit for transaction
         let gas_limit_with_overhead = tx_gas_limit
             + derive_overhead(
                 tx_gas_limit,
                 gas_per_pubdata_byte as u32,
                 tx.encoding_len(),
-                coefficients,
+                EIP_712_TX_TYPE,
```

Comment:
I am not sure you have to hardcode the type, it is available in `l2_tx`

Reply:
Just reused previous code, but good point - I will take a look